pxtlib/cpp.ts

```diff
@@ -989,6 +989,8 @@ namespace pxt.cpp {
                 .forEach(k => optSettings["YOTTA_CFG_" + k] = optSettings[k]);
         }
 
+        optSettings["PXT_TARGET"] = JSON.stringify(appTarget.id)
```

Comment:
maybe have a optional field to override this? we have platformid already.

Reply:
I think platformid is usually "codal", so maybe not so much use here